src/melee.cpp

```diff
@@ -603,8 +603,9 @@ void Character::melee_attack( Creature &t, bool allow_special, const matec_id &f
     const int encumbrance_cost = roll_remainder( ( encumb( bp_arm_l ) + encumb( bp_arm_r ) ) *
                                  2.0f );
     const int deft_bonus = hit_spread < 0 && has_trait( trait_DEFT ) ? 50 : 0;
+    const float skill_cost = std::max( 0.667f, ( 30 - melee ) / 30 );
```

Comment:
```suggestion
    const float skill_cost = std::max( 0.667f, static_cast<float>( ( 30 - melee ) / 30 ) );
```
oh well, good old float and int comparison.

Reply:
Ah right, that'd explain it!